### PR TITLE
Merged changes from development to cooler_format branch

### PR DESCRIPTION
**Pull request by joachimwolff**

### CODE REVIEW

File: hicexplorer/hicCompareMatrices.py [hunk 1]
```diff
@@ -33,161 +33,54 @@ def parse_arguments(args=None):
...
-        nan_bins = nan_bins.union(hic2.nan_bins)
-
-        if args.operation == 'diff':
-            new_matrix = hic1.matrix - hic2.matrix
-        elif args.operation == 'ratio' or args.operation == 'log2ratio':
-            hic2.matrix.data = float(1) / hic2.matrix.data
-            new_matrix = hic1.matrix.multiply(hic2.matrix)
-            # just in case
+    
+    if args.threads:
```
Comment (fidelram) [inline hicexplorer/hicCompareMatrices.py]: Since you removed the threads option from argparse, will this still work?

Comment (joachimwolff) [reply hicexplorer/hicCompareMatrices.py]: I missed to remove this from the source, thanks for the hint.

File: hicexplorer/HiCMatrix.py [hunk 2]
```diff
@@ -220,18 +235,21 @@ def load_cool(self, pMatrixFile, pChrnameList=None, pMatrixOnly=None, pIntraChro
...
                     if cut_intervals_data_frame is None:
                         cut_intervals_data_frame = cut_intervals_data_frame_
                     else:
-                        pd.concat([cut_intervals_data_frame, cut_intervals_data_frame_], ignore_index=True)
+                        pd.concat(
+                            [cut_intervals_data_frame, cut_intervals_data_frame_], ignore_index=True)
 
         else:
-            cut_intervals_data_frame = self.cooler_file.bins()[['chrom', 'start', 'end', 'weight']][:]
+            cut_intervals_data_frame = self.cooler_file.bins(
```
Comment (bgruening) [inline hicexplorer/HiCMatrix.py]: @joachimwolff you can exclude the 80char rule in linting if you like.

File: hicexplorer/hicCorrectMatrix.py [hunk 3]
```diff
@@ -577,21 +582,24 @@ def main(args=None):
         # normalize each chromosome independently
         for chrname in list(ma.interval_trees):
             chr_range = ma.getChrBinRange(chrname)
-            chr_submatrix = ma.matrix[chr_range[0]:chr_range[1], chr_range[0]:chr_range[1]]
+            chr_submatrix = ma.matrix[chr_range[0]                                      :chr_range[1], chr_range[0]:chr_range[1]]
```
Comment (bgruening) [inline hicexplorer/hicCorrectMatrix.py]: This looks odd

**Comment by bgruening** [comment]
@joachimwolff I would revert https://github.com/maxplanck-ie/HiCExplorer/pull/132/commits/5d108820d09e49708ed4cb2b26f62e9a1d02bbb4 in this PR and exclude the 80char rule. This gets hard to review.

**Comment by joachimwolff** [comment]
Don't trust default settings :) Hope it is better now.

**Comment by joachimwolff** [comment]
This is a work in progress branch, therefore I am merging it to get you the current development stage.

**Pull request merged**